src/dmd/dtemplate.d

```diff
@@ -6086,6 +6087,7 @@ extern (C++) class TemplateInstance : ScopeDsymbol
         /* Template functions may have different instantiations based on
          * "auto ref" parameters.
          */
+        if (ti.inst || ti._scope) // FIXME: hack around forward reference errors for cached TemplateInstances with inst = null
```

Comment:
This is the reason it's still a draft and needs to be ironed out. Resetting `tempinst.inst = null` after the 1st failed gagged attempt leads to problems when looking up the cached instance in the `tempdecl.instances` AA. The used `TemplateInstanceBox` key type is the only user of this `TemplaceInstance.equalsx()` method. `ti.toAlias()` requires a set `ti.inst` or `ti._scope` for a `dsymbolSemantic`, otherwise fails with a `cannot resolve forward reference` error. All of this for the following `auto ref` l/rvalue checks...